src/main/java/com/intuit/graphql/orchestrator/utils/FederationUtils.java

```diff
@@ -0,0 +1,120 @@
+package com.intuit.graphql.orchestrator.utils;
+
+import static com.intuit.graphql.orchestrator.utils.XtextTypeUtils.getFieldDefinitions;
+
+import com.intuit.graphql.graphQL.Directive;
+import com.intuit.graphql.graphQL.FieldDefinition;
+import com.intuit.graphql.graphQL.InterfaceTypeExtensionDefinition;
+import com.intuit.graphql.graphQL.ObjectTypeExtensionDefinition;
+import com.intuit.graphql.graphQL.TypeDefinition;
+import com.intuit.graphql.orchestrator.federation.exceptions.EmptyFieldsArgumentFederationDirective;
+import com.intuit.graphql.orchestrator.federation.exceptions.InvalidFieldSetReferenceException;
+import com.intuit.graphql.orchestrator.xtext.XtextGraph;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.OperationDefinition;
+import graphql.language.SelectionSet;
+import graphql.parser.Parser;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public class FederationUtils {
+    public static final String FEDERATION_KEY_DIRECTIVE = "key";
```

Comment:
should we move to FederationConstants?

Reply:
Should be in my external/refactor case pr